common-ui-compose/src/main/java/app/tivi/common/compose/Insets.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.common.compose
+
+import androidx.compose.Composable
+import androidx.compose.Providers
+import androidx.compose.StructurallyEqual
+import androidx.compose.getValue
+import androidx.compose.mutableStateOf
+import androidx.compose.onCommit
+import androidx.compose.remember
+import androidx.compose.setValue
+import androidx.compose.staticAmbientOf
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.core.ViewAmbient
+import androidx.ui.core.composed
+import androidx.ui.layout.absolutePadding
+import androidx.ui.layout.preferredHeight
+import androidx.ui.layout.preferredWidth
+import androidx.ui.unit.Density
+import androidx.ui.unit.Dp
+import androidx.ui.unit.dp
+
+/**
+ * Main holder of our inset values.
+ *
+ * TODO add other inset types (IME, visibility, etc)
+ */
+class DisplayInsets {
+    var systemBars by mutableStateOf(Insets.Zero, areEquivalent = StructurallyEqual)
+        internal set
+}
+
+data class Insets(
+    val left: Dp = 0.dp,
+    val top: Dp = 0.dp,
+    val right: Dp = 0.dp,
+    val bottom: Dp = 0.dp
+) {
+    companion object {
+        val Zero = Insets()
+    }
+}
+
+val InsetsAmbient = staticAmbientOf { DisplayInsets() }
+
+@Composable
+fun ProvideInsets(children: @Composable () -> Unit) {
+    val view = ViewAmbient.current
+    val density = DensityAmbient.current
+
+    val displayInsets = remember { DisplayInsets() }
+
+    onCommit(view.id) {
+        ViewCompat.setOnApplyWindowInsetsListener(view) { _, insets ->
+            displayInsets.updateFrom(insets, density)
+            // Return the unconsumed insets
+            insets
+        }
```

Comment:
Doesn't this need need `requestApplyInsets` here?

Reply:
Yep, added 👍